modules/actionagent/agenttool/lock.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agenttool
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/erda-project/erda/pkg/filehelper"
+)
+
+const (
+	lockFileName = "erda.lock"
+)
+
+func LockPath(path string) error {
+	return filehelper.CreateFile(filepath.Join(path, lockFileName), "", 0755)
+}
+
+func UnLockPath(path string) error {
+	lockFile := filepath.Join(path, lockFileName)
+	if err := filehelper.CheckExist(lockFile, false); err != nil {
+		return nil
+	}
+	return os.Remove(lockFile)
+}
+
+func IsPathBeLocked(path string) bool {
+	lockFile := filepath.Join(path, lockFileName)
+	if err := filehelper.CheckExist(lockFile, false); err != nil {
+		return false
+	}
+	return true
+}
+
+func WaitingPathUnlock(path string, maxLimit int) error {
+	limitTimer := time.NewTimer(time.Duration(float64(maxLimit) * float64(time.Second)))
+	ticker := time.NewTicker(time.Second)
```

Comment:
second is a too-large step. use millisecond at beginning.

Reply:
change to check locked immediately

Reply:
second still too-large.